samples/apply-unique-value-renderer/src/main/java/com/esri/arcgismaps/sample/applyuniquevaluerenderer/components/ApplyUniqueValueRendererViewModel.kt

```diff
@@ -0,0 +1,166 @@
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.FeatureLayer
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.symbology.UniqueValue
+import com.arcgismaps.mapping.symbology.UniqueValueRenderer
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel for the ApplyUniqueValueRenderer sample.
+ *
+ * This ViewModel builds an ArcGISMap, adds a FeatureLayer from a service, and
+ * applies a UniqueValueRenderer based on the SUB_REGION field. The map and
+ * layer are created during initialization and loaded in initialization. Errors are
+ * reported through message dialog.
+ */
+class ApplyUniqueValueRendererViewModel(application: Application) : AndroidViewModel(application) {
+
+    // Create the map with a Topographic basemap and an initial viewpoint
+    var arcGISMap = ArcGISMap(BasemapStyle.ArcGISTopographic).apply {
+            // Center and scale the map to show the US
+            initialViewpoint = Viewpoint(
+                center = Point(
+                    x = -12356253.6,
+                    y = 3842795.4,
+                    spatialReference = SpatialReference.webMercator()
+                ),
+                scale = 52681563.2
+            )
+    }
+
+
+    // Service feature table for the U.S. states (subregions)
+    private val censusFeatureTable = ServiceFeatureTable(
+        uri = "https://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer/3"
+    )
+
+    // Feature layer created from the service feature table
+    private val statesFeatureLayer = FeatureLayer.createWithFeatureTable(censusFeatureTable)
+
+    // Message dialog helper for presenting errors to the user
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        // Configure the renderer and add the layer to the map up-front, but load operations run in coroutine
+        configureUniqueValueRenderer()
+
+        // Add the feature layer to the map's operational layers
+        arcGISMap.operationalLayers.add(statesFeatureLayer)
+
+        // Load the map and the feature layer, report failures via the message dialog
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { throwable ->
+                messageDialogVM.showMessageDialog(
+                    title = "Failed to load map",
+                    description = throwable.message.toString()
+                )
+            }
+
+            statesFeatureLayer.load().onFailure { throwable ->
+                messageDialogVM.showMessageDialog(
+                    title = "Failed to load feature layer",
+                    description = throwable.message.toString()
+                )
+            }
```

Comment:
Optionally, you can just use the `throwable`: 
```suggestion
            arcGISMap.load().onFailure { throwable ->
                messageDialogVM.showMessageDialog(throwable)
            }

            statesFeatureLayer.load().onFailure { throwable ->
                messageDialogVM.showMessageDialog(throwable)
            }
```

Reply:
Applied separately and also removed the explicit load of the layer